[2024-06-06T22:55:47.466+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-06-06T22:55:47.483+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: weather_dag.extract_weather_data manual__2024-06-06T22:51:35.514282+00:00 [queued]>
[2024-06-06T22:55:47.490+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: weather_dag.extract_weather_data manual__2024-06-06T22:51:35.514282+00:00 [queued]>
[2024-06-06T22:55:47.492+0000] {taskinstance.py:2306} INFO - Starting attempt 3 of 6
[2024-06-06T22:55:47.502+0000] {taskinstance.py:2330} INFO - Executing <Task(SimpleHttpOperator): extract_weather_data> on 2024-06-06 22:51:35.514282+00:00
[2024-06-06T22:55:47.508+0000] {warnings.py:110} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61: DeprecationWarning: This process (pid=1447) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-06-06T22:55:47.509+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'weather_dag', 'extract_weather_data', 'manual__2024-06-06T22:51:35.514282+00:00', '--job-id', '42', '--raw', '--subdir', 'DAGS_FOLDER/weather_dag.py', '--cfg-path', '/tmp/tmpam0x07n2']
[2024-06-06T22:55:47.510+0000] {standard_task_runner.py:63} INFO - Started process 1449 to run task
[2024-06-06T22:55:47.510+0000] {standard_task_runner.py:91} INFO - Job 42: Subtask extract_weather_data
[2024-06-06T22:55:47.546+0000] {task_command.py:426} INFO - Running <TaskInstance: weather_dag.extract_weather_data manual__2024-06-06T22:51:35.514282+00:00 [running]> on host ad459dfafa58
[2024-06-06T22:55:47.605+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_EMAIL='eajavon08@gmail.com' AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='weather_dag' AIRFLOW_CTX_TASK_ID='extract_weather_data' AIRFLOW_CTX_EXECUTION_DATE='2024-06-06T22:51:35.514282+00:00' AIRFLOW_CTX_TRY_NUMBER='3' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-06-06T22:51:35.514282+00:00'
[2024-06-06T22:55:47.607+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-06-06T22:55:47.608+0000] {http.py:169} INFO - Calling HTTP method
[2024-06-06T22:55:47.614+0000] {base.py:84} INFO - Using connection ID 'weathermap_api' for task execution.
[2024-06-06T22:55:47.621+0000] {base.py:84} INFO - Using connection ID 'weathermap_api' for task execution.
[2024-06-06T22:55:49.983+0000] {http.py:208} INFO - {"coord":{"lon":-122.6762,"lat":45.5234},"weather":[{"id":801,"main":"Clouds","description":"few clouds","icon":"02d"}],"base":"stations","main":{"temp":300.12,"feels_like":299.76,"temp_min":298.07,"temp_max":302.38,"pressure":1016,"humidity":35},"visibility":10000,"wind":{"speed":4.12,"deg":310,"gust":8.75},"clouds":{"all":20},"dt":1717714082,"sys":{"type":2,"id":2004880,"country":"US","sunrise":1717676575,"sunset":1717732583},"timezone":-25200,"id":5746545,"name":"Portland","cod":200}
[2024-06-06T22:55:49.985+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-06-06T22:55:49.986+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/http/operators/http.py", line 166, in execute
    return self.execute_sync(context=context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/http/operators/http.py", line 172, in execute_sync
    return self.process_response(context=context, response=response)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/http/operators/http.py", line 215, in process_response
    return self.response_filter(response, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/weather_dag.py", line 90, in <lambda>
    response_filter = lambda r: json.load(r.text),
                                ^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/json/__init__.py", line 293, in load
    return loads(fp.read(),
                 ^^^^^^^
AttributeError: 'str' object has no attribute 'read'
[2024-06-06T22:55:50.001+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=weather_dag, task_id=extract_weather_data, run_id=manual__2024-06-06T22:51:35.514282+00:00, execution_date=20240606T225135, start_date=20240606T225547, end_date=20240606T225550
[2024-06-06T22:55:50.020+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 42 for task extract_weather_data ('str' object has no attribute 'read'; 1449)
[2024-06-06T22:55:50.062+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-06-06T22:55:50.072+0000] {warnings.py:110} WARNING - /home/***/.local/lib/python3.12/site-packages/***/models/baseoperator.py:1296: AirflowProviderDeprecationWarning: Call to deprecated class SimpleHttpOperator. (Class `SimpleHttpOperator` is deprecated and will be removed in a future release. Please use `HttpOperator` instead.)
  result = cls.__new__(cls)

[2024-06-06T22:55:50.084+0000] {taskinstance.py:3498} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-06-06T22:55:50.086+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
